text/0000-roadmap-2020.md

```diff
@@ -0,0 +1,368 @@
+
+[2019 survey]: https://blog.rust-lang.org/2019/12/03/survey-launch.html
+[rust2020]: https://readrust.net/rust-2020/
+[call for blog posts]: https://blog.rust-lang.org/2019/10/29/A-call-for-blogs-2020.html
+
+The roadmap is not meant to be "exclusive" -- that is, it's not the
+case that every single thing we do must tie in some way to the
+roadmap. But we do expect that our largest efforts will be put towards
+addressing the roadmap goals.
+
+## Structure of the roadmap
+
+The roadmap this year is based around a few central themes. These goals
+are intentionally rather broad -- they are meant to be interpreted
+throughout the year by the various teams, as they make decisions about
+what to pursue.
+
+The roadmap does not contain specific technical details or
+proposals. We encourage the individual teams to post their thoughts
+about goals and ongoing projects for 2020, either in the form of
+[Inside Rust] blog posts or as [internals] threads.
+
+[Inside Rust]: https://blog.rust-lang.org/inside-rust/index.html
+[internals]: https://internals.rust-lang.org/
+
+# The goals
+
+We have laid out three 'major goals' for Rust in 2020:
+
+* Prepare for a possible Rust 2021 Edition
+* Followthrough on in-progress designs and efforts
+* Improve project functioning and governance:
+  * Improve visibility into the state of initiatives and design efforts
+  * Increase mentoring, leadership, and organizational bandwidth
+  * Make design discussions more productive and less exhausting 
+
+## Prepare for a Rust 2021 edition
+
+[Editions] were establshed as a means to help communicate the progress of
+the Rust language and provide a rallying point for overarching pieces of work.
+One of our goals for this year should be plan out any changes that we
+wish to make as part of the next Rust edition. If we are to continue
+the three-year cadence established with the release of Rust 2018, then
+the next edition would be released in 2021.
+
+[Editions]: https://rust-lang.github.io/rfcs/2052-epochs.html
+
+One thing that we learned quite clearly from the experience of Rust
+2018 was the importance of preparation. If we wish to do a Rust 2021
+edition, we need to be planning for it now. **The goal should be that
+any changes we wish to make in Rust 2021 are completed by October of
+2020**. Completed here means that the changes are available on
+Nightly. This leaves 2021 to do tooling and polish work, such as lints
+that will port code forward.
+
+We have not yet formally decided to do an edition. **One specific scenario
+where we *would* expect to go forward with an edition is if we have work
+landed by October 2020 that relies on one.** The final decision will
+be made in October with an RFC, and it will be based on the work that
+has been completed until that date.
+
+**What might an edition contain?** We've got a number of "in progress"
+language design features that may require minor changes tied to an
+edition, but this list is by no means exhaustive:
```

Comment:
Could mention finishing `bare_trait_object` becoming a hard error & such.

Reply:
I think this RFC is should not include details as to the specific changes; it is not intended as a listing of the changes that could be made, nor is it intended as a list of the in flight changes that need an edition to be made. It's just a list that gives some ideas. I do not think adding to it would enhance this RFC's primary purpose; note that this RFC explicitly *does not* attempt to argue for or against an edition this year, merely lays out a timeline for making that decision.

Reply:
Surely the primary purpose of the paragraph is enhanced if "some ideas" is extended with plausible changes? I don't see adding more possible examples as any commitment to actually following through with them.

Reply:
I don't want to get bogged down into adding everyone's pet idea if that makes sense. I think the RFC as-is includes enough to justify discussion, adding more isn't too helpful.